src/components/datasworn/AssetCard/__tests__/AssetControl.test.tsx

```diff
@@ -74,17 +86,23 @@ describe("AssetControl", () => {
     expect(screen.getByText("Condition Meter")).toBeInTheDocument();
   });
 
-  it("should render ConditionMeter roll for condition_meter type", () => {
+  it("should roll the condition meter if condition_meter is rollable", () => {
     setup({
       control: {
         field_type: "condition_meter",
         label: "Condition Meter",
         value: 5,
         min: 0,
         max: 10,
+        rollable: true,
+        // I would add a test case for the negative, but datasworn has this hardcoded as true currently.
       } as Datasworn.AssetControlField,
     });
-    expect(screen.getByTestId("roll-button")).toBeInTheDocument();
+    expect(mockRollStat).toHaveBeenCalledTimes(0);
+    const rollButton = screen.getByTestId("roll-button");
+    expect(rollButton).toBeInTheDocument();
+    fireEvent.click(rollButton);
+    expect(mockRollStat).toHaveBeenCalledTimes(1);
```

Comment:
much better test

Reply:
Now if only it fixed the problem lol.

Reply:
I feel pretty confident that we need to make the env vars available to the build action - firebase.config is looking for them:
```const firebaseConfig: FirebaseOptions = {
  apiKey: import.meta.env.VITE_FIREBASE_APIKEY,
  authDomain: import.meta.env.VITE_FIREBASE_AUTHDOMAIN,
  projectId: import.meta.env.VITE_FIREBASE_PROJECTID,
  storageBucket: import.meta.env.VITE_FIREBASE_STORAGEBUCKET,
  messagingSenderId: import.meta.env.VITE_FIREBASE_MESSAGINGSENDERID,
  appId: import.meta.env.VITE_FIREBASE_APPID,
};```

Reply:
just kidding the module factory worked - I guess because its preventing us from getting all the way to the firebase init

Reply:
Yeah, seems like the basic vi.mock either does some kind of spy be default or imports the other modules when parsing through everything. Doing the module factory fixes that. 

Now onto the other build issue :smile: .